pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -0,0 +1,695 @@
+	if err != nil {
+		return vms
+	}
+
+	if provider == "Local" && val.BackupStore.Container != nil {
+		vms = append(vms, corev1.VolumeMount{
+			Name:      "host-storage",
+			MountPath: *val.BackupStore.Container,
+		})
+	}
+
+	if provider == "GCS" {
+		vms = append(vms, corev1.VolumeMount{
+			Name:      "etcd-backup",
+			MountPath: "/root/.gcp/",
+		})
+	} else if provider == "S3" || provider == "ABS" || provider == "OSS" || provider == "Swift" || provider == "OCS" {
+		vms = append(vms, corev1.VolumeMount{
+			Name:      "etcd-backup",
+			MountPath: "/root/etcd-backup/",
+		})
+	}
+
+	return vms
+}
+
+func getBackupRestoreVolumes(val Values) []corev1.Volume {
+	vs := []corev1.Volume{
+		{
+			Name: "etcd-config-file",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: val.ConfigMapName,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "etcd.conf.yaml",
+							Path: "etcd.conf.yaml",
+						},
+					},
+					DefaultMode: pointer.Int32(0644),
+				},
+			},
+		},
+	}
+
+	if val.ClientUrlTLS != nil {
+		vs = append(vs, corev1.Volume{
+			Name: "client-url-ca-etcd",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: val.ClientUrlTLS.TLSCASecretRef.Name,
+				},
+			},
+		},
+			corev1.Volume{
+				Name: "client-url-etcd-server-tls",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: val.ClientUrlTLS.ServerTLSSecretRef.Name,
+					},
+				},
+			},
+			corev1.Volume{
+				Name: "client-url-etcd-client-tls",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: val.ClientUrlTLS.ClientTLSSecretRef.Name,
+					},
+				},
+			})
+	}
+
+	if val.PeerUrlTLS != nil {
+		vs = append(vs, corev1.Volume{
+			Name: "peer-url-ca-etcd",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: val.PeerUrlTLS.TLSCASecretRef.Name,
+				},
+			},
+		},
+			corev1.Volume{
+				Name: "peer-url-etcd-server-tls",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: val.PeerUrlTLS.ServerTLSSecretRef.Name,
+					},
+				},
+			})
+	}
+
+	if val.BackupStore == nil {
+		return vs
+	}
+
+	storeValues := val.BackupStore
+	provider, err := utils.StorageProviderFromInfraProvider(storeValues.Provider)
+	if err != nil {
+		return vs
+	}
+
+	if provider == "Local" {
+		hpt := corev1.HostPathDirectory
+		vs = append(vs, corev1.Volume{
+			Name: "host-storage",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: defaultLocalPrefix + "/" + *storeValues.Container,
+					Type: &hpt,
+				},
+			},
+		})
+	}
+
+	if provider == "GCS" || provider == "S3" || provider == "OSS" || provider == "ABS" || provider == "Swift" || provider == "OCS" {
```

Comment:
Can you use public constants for these providers? Maybe make those (https://github.com/gardener/etcd-druid/blob/1cc064288605568386a89b1a8944b57fdd0fb31a/pkg/utils/miscellaneous.go#L52) public and use them here.

Reply:
Done